src/dataflow/analysis/tests/analysis-test.ts

```diff
@@ -28,6 +29,245 @@ function assertFailures(result: ValidationResult, expectedFailures: string[]) {
   assert.sameMembers([...result.failures], [...expectedFailuresWithoutPaths]);
 }
 
+/**
+ * Creates test nodes and edges in a chain, using the given node IDs. e.g.
+ * createChainOfEdges('A', 'B', 'C') will return an array of two edges: AB, BC.
+ */
+function createChainOfEdges(...nodeIds: string[]) {
+  assert(nodeIds.length >= 2);
+  const nodes = nodeIds.map(nodeId => new TestNode(nodeId));
+  const edges: TestEdge[] = [];
+  for (let i = 1; i < nodes.length; i++) {
+    const firstNode = nodes[i - 1];
+    const secondNode = nodes[i];
+    edges.push(new TestEdge(firstNode, secondNode, firstNode.nodeId + secondNode.nodeId));
```

Comment:
Can I request that we include a separator, for clarity. (i.e. 'A:B')

Reply:
Done